client/src/features/DataTable.js

```diff
@@ -74,14 +77,18 @@ const HeaderRow = ({ onSelectAllClick, numSelected, rowCount, data }) => {
 }
 
 
-const DataTable = ({data}) => {
+const DataTable = ({data, func}) => {
   const classes = useStyles();
-  const [selected, setSelected] = React.useState([]);
+  const [selected, setSelected] = useState([]);
+
+  useEffect(() => {
+    func(selected)
+  }, [func, selected])
 
   const handleSelectAllClick = event => {
     if (event.target.checked) {
-      const newSelecteds = data.map(n => n.Email);
-      setSelected(newSelecteds);
+      const newSelecteds = data.map(n => n.id);
+      setSelected(newSelecteds);;
```

Comment:
Was this changed to adapt to other data now, for example Campaigns?

Reply:
If you have id property on the data then yes!